src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +316,1319 @@ const ActorInit En_Egol_InitVars = {
+    s32 reaction = EYEGORE_HIT_NONE;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/EnEgol_Destroy.s")
+    if ((this->actionState == EYEGORE_ACTION_LASER) && (this->laserCollider.base.atFlags & AT_BOUNCED)) {
+        this->laserCollider.base.atFlags &= ~AT_HIT;
+        this->laserCollider.base.atFlags &= ~AT_BOUNCED;
+        this->actionTimer = 0;
+        this->laserState = EYEGORE_LASER_OFF;
+        Math_Vec3f_Copy(&this->laserScale, &gZeroVec3f);
+        Math_Vec3f_Copy(&this->laserScaleTarget, &gZeroVec3f);
+        Math_Vec3f_Copy(&this->laserCollider.dim.quad[1], &this->laserBase);
+        Math_Vec3f_Copy(&this->laserCollider.dim.quad[3], &this->laserBase);
+        Math_Vec3f_Copy(&this->laserCollider.dim.quad[0], &this->laserBase);
+        Math_Vec3f_Copy(&this->laserCollider.dim.quad[2], &this->laserBase);
+        EnEgol_SetupWalk(this);
+    }
+    if (this->eyeCollider.elements[0].info.bumperFlags & BUMP_HIT) {
+        reaction = EYEGORE_HIT_BOUNCE;
+        switch (this->actor.colChkInfo.damageEffect) {
+            case EYEGORE_DMG_LIGHT_ARROW:
+                if (ABS_ALT(angleToFacing) < 0x3000) {
+                    if ((this->actionState == EYEGORE_ACTION_STUNNED) ||
+                        (this->actionState == EYEGORE_ACTION_SLAM_WAIT)) {
+                        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, this->actor.focus.pos.x,
+                                    this->actor.focus.pos.y, this->actor.focus.pos.z, 0, 0, 0,
+                                    CLEAR_TAG_LARGE_LIGHT_RAYS);
+                        this->dmgEffectTimer = 20;
+                        reaction = EYEGORE_HIT_DAMAGE;
+                    } else if ((this->actionState >= EYEGORE_ACTION_WALK) &&
+                               (this->actionState <= EYEGORE_ACTION_STUN_END) && !this->isRetreating) {
+                        reaction = EYEGORE_HIT_NONE;
+                        this->chargingLaser = false;
+                        this->chargeLevel = 0;
+                        this->actionTimer = 0;
+                        this->laserState = EYEGORE_LASER_OFF;
+                        Math_Vec3f_Copy(&this->laserScale, &gZeroVec3f);
+                        Math_Vec3f_Copy(&this->laserScaleTarget, &gZeroVec3f);
+                        Math_Vec3f_Copy(&this->laserCollider.dim.quad[1], &this->laserBase);
+                        Math_Vec3f_Copy(&this->laserCollider.dim.quad[3], &this->laserBase);
+                        Math_Vec3f_Copy(&this->laserCollider.dim.quad[0], &this->laserBase);
+                        Math_Vec3f_Copy(&this->laserCollider.dim.quad[2], &this->laserBase);
+                        Actor_SetColorFilter(&this->actor, 0, 120, false, 40);
+                        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_COMMON_FREEZE);
+                        EnEgol_SetupStunned(this);
+                    }
+                }
+                break;
+            case EYEGORE_DMG_NORMAL:
```

Comment:
```suggestion
                break;

            case EYEGORE_DMG_NORMAL:
```
Makes it clearer it's not a fallthrough.